Appveyor で MinGW ビルドが失敗する

## 問題内容
Appveyor で MinGW ビルドが失敗する

https://ci.appveyor.com/project/sakuraeditor/sakura/build/job/qiic2dej9f2dbyg7

```
79: g++ -finput-charset=utf-8 -fexec-charset=cp932 -I. -DWIN32 -D_WIN32_WINNT=_WIN32_WINNT_WIN7 -D_UNICODE -DUNICODE -DNDEBUG -O2  ../HeaderMake/HeaderMake.cpp -o ../HeaderMake/HeaderMake.exe -static-libgcc
80: ../HeaderMake/HeaderMake.cpp:80:6: error: ambiguating new declaration of 'void fopen_s(FILE**, const char*, const char*)'
81:   80 | void fopen_s(
82:      |      ^~~~~~~
83: In file included from C:/msys64/mingw64/x86_64-w64-mingw32/include/stdio.h:1395,
84:                  from ../HeaderMake/HeaderMake.cpp:57:
85: C:/msys64/mingw64/x86_64-w64-mingw32/include/sec_api/stdio_s.h:541:27: note: old declaration 'errno_t fopen_s(FILE**, const char*, const char*)'
86:   541 |   _CRTIMP errno_t __cdecl fopen_s(FILE **_File,const char *_Filename,const char *_Mode);
87:       |                           ^~~~~~~
88: mingw32-make: *** [Makefile:458: ../HeaderMake/HeaderMake.exe] Error 1
89: mingw32-make: Leaving directory 'C:/projects/sakura/sakura_core'
90: error 1 errorlevel 2
91: error build-gnu.bat       
92: Command exited with code 1
```

https://ci.appveyor.com/project/sakuraeditor/sakura/builds/26404372
ではビルドに成功している

ビルド環境が変わった?

## 再現手順
Appveyor で MinGW ビルドする

## 再現頻度
100%

## 問題のカテゴリ

- ビルドの問題
  - Appveyor

## 環境情報
Appveyor 

## スクリーンショット
<!-- 説明に必要なスクリーンショットがあれば貼り付けお願いします。-->
<!-- 画像ファイルをこの欄にドラッグ＆ドロップすれば画像が貼り付けられます -->

--- a/HeaderMake/HeaderMake.cpp
+++ b/HeaderMake/HeaderMake.cpp
@@ -47,13 +47,6 @@
 	usage() を参照
 */
 
-#ifdef __MINGW32__
-#include <_mingw.h>
-#ifdef MINGW_HAS_SECURE_API
-#undef MINGW_HAS_SECURE_API
-#endif  // MINGW_HAS_SECURE_API
-#endif  // __MINGW32__
-
 #include <stdio.h>
 #include <string>
 #include <vector>
@@ -64,38 +57,10 @@
 #include <errno.h>
 using namespace std;
 
-#define PREPROCESSOR "cl.exe /nologo /source-charset:utf-8 /execution-charset:shift_jis /EP %s"
-
-#ifdef __MINGW32__
-#include <windows.h>
-#ifndef _countof
-#define _countof(A) (sizeof(A)/sizeof(A[0]))
-#endif
-#define sprintf_s(A, B, C, ...) sprintf((A), (C), (__VA_ARGS__))
-#define strncpy_s(A, B, C, D) strncpy((A), (C), (D))
-
-#undef PREPROCESSOR
-#define PREPROCESSOR "gcc -x c++ -finput-charset=utf-8 -fexec-charset=cp932 -E %s"
-
-void fopen_s( 
-   FILE** pFile,
-   const char *filename,
-   const char *mode 
-)
-{
-	*pFile = fopen(filename, mode);
-}
-#endif	// __MINGW32__
-
 #ifdef _MSC_VER
-#if _MSC_VER < 1400	// VC2003
-#ifndef _countof
-#define _countof(A) (sizeof(A)/sizeof(A[0]))
-#endif
-#define sprintf_s(A, B, C, D) sprintf((A), (C), (D))
-#define strncpy_s(A, B, C, D) strncpy((A), (C), (D))
-#define fopen_s(A, B, C) ( *(A) = fopen((B), (C)) )
-#endif	// VC2003
+#define PREPROCESSOR "cl.exe /nologo /source-charset:utf-8 /execution-charset:shift_jis /EP %s"
+#elif defined(__GNUC__)
+#define PREPROCESSOR "gcc.exe -x c++ -finput-charset=utf-8 -fexec-charset=cp932 -E %s"
 #endif	// _MSC_VER
 
 enum EMode{

